src/store/TodoListProvider.tsx

```diff
@@ -0,0 +1,42 @@
+import { useState, createContext, ReactNode } from 'react';
+import { getTodoList } from '@todo/api';
+import { TodoItem } from './TodoItem';
+
+type TodoListQuery = {
+  isLoading: boolean;
+  error?: unknown | undefined;
+  todoList: ReadonlyArray<TodoItem>;
+};
+
+type TodoListQueryProps = {
+  queryProps: TodoListQuery;
+  queryTodoList: () => Promise<void>;
+};
+
+const TodoListContext = createContext<TodoListQueryProps>({
+  queryProps: { isLoading: false, error: '', todoList: [] },
+  queryTodoList: async () => {},
+});
+
+const TodoListProvider = ({ children }: { children: ReactNode }) => {
+  const [queryProps, setQueryProps] = useState<TodoListQuery>({
+    isLoading: false,
+    error: '',
+    todoList: [],
+  });
+
+  const queryTodoList = async () => {
+    setQueryProps({ isLoading: true, error: undefined, todoList: [] });
+    try {
+      const response = await getTodoList();
+      const data = await response.json();
+      setQueryProps({ isLoading: false, error: '', todoList: data });
```

Comment:
`error: undefined` (`error` isn't a `string`)

Reply:
I think, this is extra. Deleted.